### PR TITLE
Add reflectance output  (feature/crtm_add_reflectance_output)

### LINKED ISSUE #84: Add TOA albedo/reflectance in CRTM
Add TOA albedo/reflectance in CRTM forward/TL/AD calculations

- [x] Merge initial implementation from v2.4.1 into v3.
- [ ] Add TL/AD calculations.
- [ ] Add forward/TL/AD unit tests.
- [x] Add TOA reflectance in all RTS I/O interfaces.
- [ ] Offline scientific tests for computed forward/TL/AD calculations.

### PR DESCRIPTION
**Pull request by BenjaminTJohnson**
## Description

Add visible reflectance output to CRTM in support of visible sensor assimilation.  

## Issue(s) addressed

Resolves #84 #95 #69 

## Dependencies

List the other PRs that this PR is dependent on:
- [x] #83 

## Impact

Expected impact on downstream repositories:
N/A

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

File: src/RTSolution/Common_RTSolution.f90 [hunk 1]
```diff
@@ -1232,6 +1232,16 @@ FUNCTION Assign_Common_Output( &
...
              RTSolution%Radiance,              & ! Input
              RTSolution%Brightness_Temperature ) ! Output  
+
+    ! -------------------------------------------------------
+    ! Compute the corresponding solar reflectance (TOA only)
+    ! -------------------------------------------------------
+    RTSolution%Reflectance = ZERO
+    IF( RTV%Solar_Flag_true .AND. (.NOT.(RTV%aircraft%rt))) THEN
+         RTSolution%Reflectance = RTSolution%Radiance*PI/RTV%Solar_irradiance
+         !  write(*, '(a,f9.4,a,f7.4,a,f7.4)') ' DEBUG-GREG, Solar_irradiance, reflectance, cos_sun = ', &
```
Comment (gthompsnJCSDA) [inline src/RTSolution/Common_RTSolution.f90]: You may want to remove my DEBUG prints.

File: src/RTSolution/CRTM_RTSolution.f90 [hunk 2]
```diff
@@ -370,6 +370,9 @@ FUNCTION CRTM_Compute_RTSolution( &
       RETURN
     END IF
 
+    !DO i = 1, Atmosphere%n_Layers
```
Comment (gthompsnJCSDA) [inline src/RTSolution/CRTM_RTSolution.f90]: You may want to remove my debug print

**Review by gthompsnJCSDA — APPROVED** [review]

**Comment by chengdang** [comment]
Is this meant to be a draft branch?

This branch is not compatible with the current dev branch. In CRTM_RTSolution_Define.f90, few updates are needed, at the first glance:
- `CRTM_RTSolution_WriteFile_NetCDF `has been updated to generate one file per sensor.
- TOA reflectance need to be added to RTS read modules.

I can push these updates later this week.

Meanwhile, TL/K-matrix calculations are not completed as we discussed earlier.

**Comment by BenjaminTJohnson** [comment]
@chengdang converted to draft while remaining tasks are being worked.

**Comment by BenjaminTJohnson** [comment]
@chengdang yes, there's some weird stuff here -- I'm confused why RTSolution has changed structurally.   As it stands right now, I can't get it to compile -- I'll leave it alone and let you work on it.

**Comment by gthompsnJCSDA** [comment]
@chengdang  I just incorporated changes intended to get the clear-sky reflectance similar to having radiance in cloudy as well as clear skies. The final calculation just changes to
```
  Reflectance_clear = RTSolution%R_clear * PI / RTSolution%SolarIrradiance
```
since we already have the clear sky radiance in the structure.  So I added the new `Reflectance_clear` to the RTSolution structure and all the associated initialization, read/write places, etc.  I did so in a manner that matches how `R_clear` and `Tb_clear` are done.  Lastly, I don't think we need to bother with the adjoint because R_clear and Tb_clear are not part of the adjoint.

**Comment by chengdang** [comment]
@gthompsnJCSDA Thanks Greg! The updates look good to me.

I just merged the dev branch in, and updated `CRTM_RTSolution_Define.f90`, for the following features:
- `CRTM_RTSolution_WriteFile_NetCDF` has been updated to generate one file per sensor.
- TOA reflectance has been added to RTS read modules.

**Review by chengdang — APPROVED** [review]

**Comment by chengdang** [comment]
@BenjaminTJohnson 
Hello Ben, I pushed another commit to remove the reflectance calculation in `FUNCTION Assign_Common_Output_AD`, as I realized we need to update all AD related variables, including atm_K, sfc_K, and rts_k for the AD reflectance implementation to work.

All code updates in this PR are for forward calculation only, which have been tested and reviewed.

If it looks good, shall we merge this PR first? I'll draft another one with AD implementation for our discussion. @gthompsnJCSDA


The following code block is commented out in `FUNCTION Assign_Common_Output_AD`
```
    ! ----------------------------------------------------------------
    ! Compute the corresponding adjoint solar reflectance (TOA only)
    ! ----------------------------------------------------------------
    ! RTSolution_AD%Reflectance = ZERO
    ! IF( RTV%Solar_Flag_true .AND. (.NOT.(RTV%aircraft%rt))) THEN
    !  RTSolution_AD%Reflectance = RTSolution_AD%Radiance*PI/RTV%Solar_irradiance
    ! END IF
```

**Review by chengdang — APPROVED** [review]

**Pull request merged**